packages/commercetools/helpers/src/index.ts

```diff
@@ -24,30 +27,64 @@ export const getProductGallery = (product: ProductVariant): UiMediaGalleryItem[]
   }))
 
 /** Returns array of product variants meeting criteria */
-export const getProductVariants = (products: ProductVariant[], options: any = {}): ProductVariant | ProductVariant[]  => {
+export const getProductVariants = (products: ProductVariant[], filters: any = {}): ProductVariant | ProductVariant[]  => {
   if (!products) {
     return []
   }
 
-  if (options.master) {
+  if (filters.attributes && Object.keys(filters.attributes).length > 0) {
+    return getVariantByAttributes(products, filters.attributes)
+  }
+
+  if (filters.master) {
     return products.find(product => (product as any)._master)
   }
 
   return products
 }
 
-export const getProductAttributes = (product: ProductVariant, filterByAttributeName?: Array<string>): Array<AgnosticProductAttribute> => {
-  return (product ? formatAttributeList(product.attributeList) : [])
-  .filter(attribute => {
-    if (filterByAttributeName) return filterByAttributeName.includes(attribute.name)
-    return attribute
+export const getProductAttributes = <A extends string>(products: ProductVariant[], filterByAttributeName?: Array<string>): AgnosticProductOptions<A> => {
```

Comment:
why this weird typing

Reply:
also, imho it's better to have `AgnosticProductOption[]` instead of plural typing